framework/validators/FileValidator.php

```diff
@@ -420,4 +420,37 @@ protected function getClientOptions($model, $attribute)
 
         return $options;
     }
+
+    /**
+     * @param $mimeType
+     * @param $mask
+     * @return bool
+     */
+    protected function validateMimeTypeMask($mimeType, $mask)
+    {
+        $regexp = "/^" . str_replace('\*', '.*', preg_quote($mask, "/")) . "$/";
+        return (bool)preg_match($regexp, $mimeType);
+    }
+
+    /**
+     * @param UploadedFile $file
+     * @return bool
+     * @throws \yii\base\InvalidConfigException
+     */
+    protected function validateMimeType($file)
+    {
+        $fileMimeType = FileHelper::getMimeType($file->tempName);
+
+        foreach ($this->mimeTypes as $mimeType) {
+            if (strpos($mimeType, '*') !== false && $this->validateMimeTypeMask($fileMimeType, $mimeType)) {
+                return true;
+            }
+
+            if ($mimeType == $fileMimeType) {
```

Comment:
Since the condition is simpler than the one above, this one could be checked first.